src/authentication/UserAuthenticator.js

```diff
@@ -55,7 +57,7 @@ export class UserAuthenticator {
       // Race condition promise, if connection is taking to long will be rejected
       // with CONNECTION_FAILURE error
       const userJson = await Promise.race([
-        authenticateAsync(authURL, username, passwordHash),
+        authenticateAsync(authURL, username, passwordHash, uniqueId),
```

Comment:
@andreievg This may also leads to do changes in other places like `GET, POST` method inside `Synchroniser.js` file? Do you think the above change (passing extra header object rather than a single header text `msupply-site-uuid`) should also require inside `Synchroniser.js` rest api methods? If yes then I may also need to update on 4D mSupply backend part to parse and make readable for site uuid to check against the database.

Reply:
@rabiSussol, i mean to write it this way, sorry:

`authenticateAsync(authURL, username, passwordHash, {'msupply-site-uuid':uniquID})`

or even better (since uniqueId should be stored in settings)

```
constructor(database, settings) {
this.extraHeaders = {'msupply-site-uuid': settings.get(HARDWARE_UUID)};
....
...
....
authenticateAsync(authURL, username, passwordHash, this.extraHeaders)
```

same for Synchroniser: 
```
headers: {
   Authorization: this.authHeader,
   ...this.extraHeaders
}

```
what do you think ?